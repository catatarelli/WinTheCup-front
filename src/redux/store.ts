import {
  configureStore,
  type ThunkAction,
  type Action,
} from "@reduxjs/toolkit";
import { uiReducer } from "./features/uiSlice/uiSlice";
import { userReducer } from "./features/user/userSlice";

export const store = configureStore({
  reducer: {
    ui: uiReducer,
    user: userReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
