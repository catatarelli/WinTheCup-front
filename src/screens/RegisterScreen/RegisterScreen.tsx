/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import { ImageBackground, SafeAreaView, ScrollView, Image } from "react-native";
import RegisterForm from "../../components/RegisterForm/RegisterForm";
import React from "react";
import screenStyles from "../../styles/screen.styles";
import image from "../../../assets/background-image.webp";
import logo from "../../../assets/logo.png";
import formStyles from "../../styles/form.styles";

const RegisterScreen = () => (
  <ScrollView>
    <ImageBackground
      source={image}
      resizeMode="cover"
      style={screenStyles.image}
      testID="backgroudImage"
    >
      <SafeAreaView style={screenStyles.container}>
        <Image source={logo} style={formStyles.logoForm} testID="logo" />
        <RegisterForm />
      </SafeAreaView>
    </ImageBackground>
  </ScrollView>
);

export default RegisterScreen;
