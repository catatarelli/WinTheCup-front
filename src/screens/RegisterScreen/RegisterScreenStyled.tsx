import { Dimensions, StyleSheet } from "react-native";
import colors from "../../styles/colors.styles";

const screenHeight = Dimensions.get("window").height;
const screenWidth = Dimensions.get("window").width;

const styles = StyleSheet.create({
  container: {
    display: "flex",
    alignItems: "center",
  },
  image: {
    height: screenHeight,
    width: screenWidth,
    justifyContent: "center",
    alignItems: "center",
  },
});

export default styles;
