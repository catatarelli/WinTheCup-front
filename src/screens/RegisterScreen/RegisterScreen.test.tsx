import React from "react";
import { render, screen } from "@testing-library/react-native";
import RegisterScreen from "./RegisterScreen";
import { Provider } from "react-redux";
import { store } from "../../redux/store";

describe("Given a RegisterScreen component", () => {
  describe("When it's rendered", () => {
    test("Then it should show a background image", async () => {
      const imageId = "backgroudImage";

      render(
        <Provider store={store}>
          <RegisterScreen />
        </Provider>
      );

      const displayedImage = screen.queryByTestId(imageId);

      expect(displayedImage).toBeDefined();
    });
  });
});
