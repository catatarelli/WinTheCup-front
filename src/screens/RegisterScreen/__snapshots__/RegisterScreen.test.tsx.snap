// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a RegisterScreen component When it's rendered Then it should match the snapshot 1`] = `
<RCTScrollView>
  <View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 1334,
          "justifyContent": "center",
          "resizeMode": "cover",
          "width": 750,
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 1334,
              "width": 750,
            },
            undefined,
          ]
        }
        testID="backgroudImage"
      />
      <RCTSafeAreaView
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 46,
              "marginBottom": 50,
              "marginTop": 100,
              "width": 350,
            }
          }
          testID="logo"
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#000",
              "display": "flex",
              "height": 551,
              "opacity": 0.7,
              "padding": 15,
              "width": 375,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 30,
                }
              }
            >
              Username
            </Text>
            <TextInput
              maxLength={20}
              onChangeText={[Function]}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#fff",
                  "borderStyle": "solid",
                  "borderWidth": 2,
                  "color": "#3F3F3F",
                  "fontSize": 26,
                  "height": 54,
                  "marginVertical": 10,
                  "padding": 10,
                  "width": 300,
                }
              }
              testID="username"
              textContentType="username"
              value=""
            />
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 30,
                }
              }
            >
              Password
            </Text>
            <TextInput
              maxLength={20}
              onChangeText={[Function]}
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#fff",
                  "borderStyle": "solid",
                  "borderWidth": 2,
                  "color": "#3F3F3F",
                  "fontSize": 26,
                  "height": 54,
                  "marginVertical": 10,
                  "padding": 10,
                  "width": 300,
                }
              }
              testID="password"
              textContentType="password"
              value=""
            />
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 30,
                }
              }
            >
              Email
            </Text>
            <TextInput
              accessibilityLabel="email"
              maxLength={25}
              onChangeText={[Function]}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#fff",
                  "borderStyle": "solid",
                  "borderWidth": 2,
                  "color": "#3F3F3F",
                  "fontSize": 26,
                  "height": 54,
                  "marginVertical": 10,
                  "padding": 10,
                  "width": 300,
                }
              }
              testID="email"
              textContentType="emailAddress"
              value=""
            />
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 20,
                "color": "#fff",
                "height": 57,
                "justifyContent": "center",
                "marginVertical": 25,
                "opacity": 0.3,
                "padding": 10,
                "width": 266,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "fontSize": 30,
                  "fontWeight": "500",
                  "zIndex": 1,
                }
              }
              testID="submitButton"
            >
              Join Now
            </Text>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                  "marginRight": 15,
                }
              }
            >
              Have an account?
            </Text>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Log in
              </Text>
            </View>
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
  </View>
</RCTScrollView>
`;
